Pod/Classes/Cocoa+CMHealth.m

```diff
@@ -72,4 +72,79 @@ - (instancetype)initWithCoder_cmh:(NSCoder *)decoder
     return [self initWithCoder_cmh:decoder];
 }
 
+@end
+
+@implementation NSCalendar (CMHealth)
+
++ (void)load {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        cmh_swizzle([self class], @selector(initWithCoder:), @selector(initWithCoder_cmh:));
+        cmh_swizzle([self class], @selector(encodeWithCoder:), @selector(cmh_encodeWithCoder:));
+    });
+}
+
+- (void)cmh_encodeWithCoder:(NSCoder *)aCoder
+{
+    if ([aCoder isKindOfClass:[CMObjectEncoder class]]) {   
+        [aCoder encodeObject:self.calendarIdentifier forKey:@"calendarIdentifier"];
+        [aCoder encodeObject:self.locale forKey:@"locale"];
+        [aCoder encodeObject:self.timeZone forKey:@"timeZone"];
+        [aCoder encodeInteger:self.firstWeekday forKey:@"firstWeekday"];
+        [aCoder encodeInteger:self.minimumDaysInFirstWeek forKey:@"minimumDaysInFirstWeek"];
+        return;
+    }
+
+    [self cmh_encodeWithCoder:aCoder];
+}
+
+- (instancetype)initWithCoder_cmh:(NSCoder *)decoder
+{
+    if ([decoder isKindOfClass:[CMObjectDecoder class]]) {
+        self = [[NSCalendar alloc] initWithCalendarIdentifier:[decoder decodeObjectForKey:@"calendarIdentifier"]];
+        self.locale = [decoder decodeObjectForKey:@"locale"];
+        self.timeZone = [decoder decodeObjectForKey:@"timeZone"];
+        self.firstWeekday = [decoder decodeIntegerForKey:@"firstWeekday"];
+        self.minimumDaysInFirstWeek = [decoder decodeIntegerForKey:@"minimumDaysInFirstWeek"];
+        return self;
+    }
+
+    return [self initWithCoder_cmh:decoder];
+}
+
+@end
+
+@implementation NSTimeZone (CMHealth)
+
++ (void)load {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        cmh_swizzle([self class], @selector(initWithCoder:), @selector(initWithCoder_cmh:));
+        cmh_swizzle([self class], @selector(encodeWithCoder:), @selector(cmh_encodeWithCoder:));
+    });
+}
+
+- (void)cmh_encodeWithCoder:(NSCoder *)aCoder
+{
+    if ([aCoder isKindOfClass:[CMObjectEncoder class]]) {
+        [aCoder encodeObject:self.name forKey:@"name"];
+        return;
+    }
+
+    [self cmh_encodeWithCoder:aCoder];
+}
+
+- (instancetype)initWithCoder_cmh:(NSCoder *)decoder
```

Comment:
By convention, ObjC developer generally only use underscores in specific cases, usually to prefix something being added by a third party, such as a inside a Category provided by a library. In those instances, the prefixes both denote the third-party nature of the method and also prevent namespace collisions.

In this case, the methods you see are being used to **replace** existing methods on Foundation classes at runtime (typically called Swizzling, or Monkey Patching). Therefore they have to have a different name, but it's useful to have the name be related to the method it will ultimately replace.

In the case of the postfix, this is because we are Swizzling an `init` method. `Init` methods in ObjC must start with `init`, hence a postfix is used instead. I agree it's ugly, but I think it's the lease ugly, most explicit option in this case. The good news is that these methods are totally internal to the SDK. Developers will use them, but without knowing, and without having to call the ugly method names.